apps/client/app/components/PurchaseOrderForm.tsx

```diff
@@ -0,0 +1,172 @@
+// This is the only client component so far in the application due to the need for interactivity
+'use client'
+
+import React, { useEffect, useState } from 'react';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { PurchaseOrder, Item } from '../interfaces';
+
+// This component is passed in the Purchase Order and whether we are creating or editing
+interface PurchaseOrderFormProps {
+  purchaseOrder?: PurchaseOrder;
+  items: Item[];
+  mode: 'create' | 'edit';
+}
+
+// Define the validation schema using Yup
+const validationSchema = Yup.object().shape({
+  id: Yup.number().required('Purchase Order is required'),
+  vendor_name: Yup.string().required('Vendor name is required'),
+  order_date: Yup.string().required('Order date is required'),
+  expected_delivery_date: Yup.string().required('Expected delivery date is required').test(
+    'is-greater',
+    'Expected delivery date must be after the order date',
+    function (value) {
+      const { order_date } = this.parent;
+      return new Date(order_date) < new Date(value);
+    }
+  ),
+  purchase_order_line_items: Yup.array().of(
+    Yup.object().shape({
+      id: Yup.number().required('Purchase Order Line Item is required'),
+      item_id: Yup.number().required('Item is required').positive(),
+      quantity: Yup.number().required('Quantity is required').positive().integer(),
+      unit_cost: Yup.number().required('Unit cost is required').positive(),
+    })
+  ).required(),
+  total_quantity: Yup.number().notRequired(),
+  total_cost: Yup.number().notRequired(),
+});
+
+const PurchaseOrderForm: React.FC<PurchaseOrderFormProps> = ({ purchaseOrder, items, mode }) => {
+  const [serverErrors, setServerErrors] = useState<string[]>([]);
+
+  // Initialize the form with react-hook-form
+  const { control, handleSubmit, formState: { errors } } = useForm<PurchaseOrder>({
+    defaultValues: {
+      id: purchaseOrder?.id || 0,
+      vendor_name: purchaseOrder?.vendor_name || '',
+      order_date: purchaseOrder?.order_date.substring(0, 10) || '',
+      expected_delivery_date: purchaseOrder?.expected_delivery_date.substring(0, 10) || '',
+      purchase_order_line_items: purchaseOrder?.purchase_order_line_items.map(lineItem => ({
+        id: lineItem.id || 0,
+        item_id: lineItem.item_id,
+        quantity: lineItem.quantity,
+        unit_cost: lineItem.unit_cost,
+      })) || [{ id: 0, item_id: 0, quantity: 0, unit_cost: 0 }]
+    },
+    resolver: yupResolver(validationSchema) as any, // Casting to any to get rid of TypeScript error that is not actually an issue
```

Comment:
There's probably a rhf value to cast to here, but yeah ts and RHF don't always work that well together. especially when you start using watch()